src/py-opentimelineio/opentimelineio/adapters/svg.py

```diff
@@ -138,207 +205,347 @@ def contract(self, distance):
-               r' style="stroke:{};stroke-width:{};' \
-               r'fill:{};opacity:{};">{}</text>'.format(repr(text_size),
-                                                        font_family,
-                                                        repr(location_svg.x),
-                                                        repr(location_svg.y),
-                                                        svg_color(color),
-                                                        repr(stroke_width / 4.0),
-                                                        svg_color(color),
-                                                        repr(color[3]),
-                                                        text)
-    lines.append(text_str)
-
-
-def get_text_layout_size(text='', text_size=10.0):
-    text_width = 0.0
-    for character in text:
-        ascii_val = int(ord(character))
-        text_width += LCARS_CHAR_SIZE_ARRAY[ascii_val] if ascii_val < 128 else 0
-    scale_factor = text_size * 0.01
-    return text_width * scale_factor + 25
-
-
-def get_image():
-    header = r'<svg height="{}" width="{}" version="4.0"' \
-             r' xmlns="http://www.w3.org/2000/svg"' \
-             r' xmlns:xlink= "http://www.w3.org/1999/xlink">'.format(image_height,
-                                                                     image_width)
-    background = r'<rect width="100%" height="100%" fill="white"/>'
-    font = r'<defs><style>@import url("https://fonts.googleapis.com/css?' \
-           r'family={}");</style></defs>'.format(font_family)
-    separator = '\n'
-    image = header + '\n' + background + '\n' + font + '\n' + separator.join(
-        lines) + '\n</svg>'
-    return image
+              text_size, color=COLORS['black'], stroke_width=1.0, config=None):
+    location_svg = convert_point_to_svg_coordinates(location, config=config)
+    text_elem = ET.SubElement(config.svg_elem, "text",
+                              OrderedDict([
+                                  ("x", repr(location_svg.x)),
+                                  ("y", repr(location_svg.y)),
+                                  ("font-size", repr(text_size)),
+                                  ("font-family", config.font_family),
+                                  ("style", "stroke:{};stroke-width:{};"
+                                            "fill:{};opacity:{};"
+                                   .format(Color.svg_color(color),
+                                           repr(stroke_width / 4.0),
+                                           Color.svg_color(color),
+                                           repr(color[3] / 255.0)))
+                              ]),
+                              )
+    text_elem.text = text
+
+
+def get_image(config):
+    xmlstr = ET.tostring(config.svg_elem, encoding='utf-8', method='xml')
+    try:
+        xmlstr = xmlstr.decode("utf8")
+    except (UnicodeDecodeError, AttributeError):
+        pass
```

Comment:
Python 3 produces a byte string with the tostring() method, but python 2 gives an ate object. This pattern here is to check for this case. I'll add a comment.

Reply:
Ahh, makes sense. It seems like in that case it would always throw the same exception, why are there two different exception types handled?

In the case of bytes (py 3) vs. bytestring (py 2), using `isinstance(xmlstr, bytes)` might be the more explicit way to branch. Either way, I'd still include some comments about it.

Reply:
I thought, what if xmlstr was not a bytestring or string for some reason. But that doesn't make sense now :sweat_smile:. I should just be checking for UnicodeDecodeError.

py3 gives bytestring, and py2 gives str. Will `isinstance(xmlstr, bytes)` work for bytestring?

Reply:
Are you sure about `bytestring`?

```
$ python3
Python 3.8.6 (default, Nov 12 2020, 12:54:53)
[Clang 12.0.0 (clang-1200.0.32.21)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> bytestring
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'bytestring' is not defined
>>> from xml.etree.ElementTree import Element, SubElement, tostring
>>> elem = Element("rootElem")
>>> xmlstr = tostring(elem, encoding='utf-8', method='xml')
>>> type(xmlstr)
<class 'bytes'>
>>>
$ python2
Python 2.7.17 (default, Oct 24 2019, 12:57:47)
[GCC 4.2.1 Compatible Apple LLVM 11.0.0 (clang-1100.0.33.8)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from xml.etree.ElementTree import Element, SubElement, tostring
>>> elem = Element("rootElem")
>>> xmlstr = tostring(elem, encoding='utf-8', method='xml')
>>> type(xmlstr)
<type 'str'>
```

Reply:
Ohh. I didn't know str was called byte string. When I printed the result from py3 it gave me something like b'abcd'.